jenkins.py

```diff
@@ -22,6 +22,20 @@
 PLUGIN_NAME = "jenkins"
 DEFAULT_API_TIMEOUT = 60
 
+
+def build_job_folding_exp(depth):
+    """
+    Builds an expression to get all folding jobs within specified depth
+    """
+    exp = ""
+    for _ in range(depth):
+        exp = "[jobs" + exp + ",name,url]"
+    return exp
+
+
+JOB_FOLDING_MAX_DEPTH = 5
```

Comment:
Any logic behind 5?

Reply:
We wanted to make a single api request to retrieve jobs.  There is not a single Jenkins rest api endpoint that can do this, but you can request a tree of a certain depth.  We picked an arbitrary depth of 5 which we thought would cover most cases, but this could be increased to an arbitrarily large depth eg. 100.

One concern is that with a large number of nested jobs, the python plugin could timeout before all jobs are processed.

Reply:
Yea i was actually thinking the other way around. That 5 was quite deep. Regardless of the depth its probably more appropriate to scope on job count, although that's probably out of scope here.